src/main/scala/ai/privado/languageEngine/java/language/package.scala

```diff
@@ -69,4 +69,8 @@ package object language {
       None
     }
   }
+
+  implicit class NodeToOriginalSource(val trav: Traversal[AstNode]) extends AnyVal {
+    def originalSource: Traversal[AstNode] = trav.out(EdgeTypes.ORIGINAL_SOURCE).cast[AstNode]
```

Comment:
Instead of `cast` better to use `collect` or `collectAll`

Reply:
Similar to `originalSource`, we should also create a function `derivedSource` which we can use to get all derived sources generated from a node